src/signalrclient/json_helpers.cpp

```diff
@@ -3,78 +3,104 @@
 
 #include "stdafx.h"
 #include "json_helpers.h"
+#include <cmath>
 
 namespace signalr
 {
-    signalr::value createValue(const web::json::value& v)
+    signalr::value createValue(const Json::Value& v)
     {
         switch (v.type())
         {
-        case web::json::value::Boolean:
-            return signalr::value(v.as_bool());
-        case web::json::value::Number:
-            return signalr::value(v.as_double());
-        case web::json::value::String:
-            return signalr::value(utility::conversions::to_utf8string(v.as_string()));
-        case web::json::value::Array:
+        case Json::ValueType::booleanValue:
+            return signalr::value(v.asBool());
+        case Json::ValueType::realValue:
+        case Json::ValueType::intValue:
+        case Json::ValueType::uintValue:
+            return signalr::value(v.asDouble());
+        case Json::ValueType::stringValue:
+            return signalr::value(v.asString());
+        case Json::ValueType::arrayValue:
         {
-            auto& array = v.as_array();
             std::vector<signalr::value> vec;
-            for (auto& val : array)
+            for (auto& val : v)
             {
                 vec.push_back(createValue(val));
             }
             return signalr::value(std::move(vec));
         }
-        case web::json::value::Object:
+        case Json::ValueType::objectValue:
         {
-            auto& obj = v.as_object();
             std::map<std::string, signalr::value> map;
-            for (auto& val : obj)
+            for (const auto& val : v.getMemberNames())
             {
-                map.insert({ utility::conversions::to_utf8string(val.first), createValue(val.second) });
+                map.insert({ val, createValue(v[val]) });
             }
             return signalr::value(std::move(map));
         }
-        case web::json::value::Null:
+        case Json::ValueType::nullValue:
         default:
             return signalr::value();
         }
     }
 
-    web::json::value createJson(const signalr::value& v)
+    Json::Value createJson(const signalr::value& v)
     {
         switch (v.type())
         {
         case signalr::value_type::boolean:
-            return web::json::value(v.as_bool());
+            return Json::Value(v.as_bool());
         case signalr::value_type::float64:
-            return web::json::value(v.as_double());
+        {
+            auto value = v.as_double();
+            double intPart;
+            // REVIEW: workaround for 1.0 being output as 1.0 instead of 1
+            if (std::modf(value, &intPart) == 0)
```

Comment:
This seems reasonable to me. Why the "REVIEW" comment?

Reply:
Wanted to call attention to it.

There are alternatives, but I'm fairly ok with doing it this way.